File: daniel-alves-moura/nexuserp/backend/src/main/java/br/com/nexuserp/service/AuthService.java
package br.com.nexuserp.service;

import br.com.nexuserp.dto.auth.*;
import br.com.nexuserp.entity.Role;
import br.com.nexuserp.entity.User;
import br.com.nexuserp.exception.BusinessException;
import br.com.nexuserp.repository.UserRepository;
import br.com.nexuserp.security.JwtService;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
public class AuthService {
    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final AuthenticationManager authenticationManager;
    private final JwtService jwtService;

    public AuthService(UserRepository userRepository, PasswordEncoder passwordEncoder,
                       AuthenticationManager authenticationManager, JwtService jwtService) {
        this.userRepository = userRepository;
        this.passwordEncoder = passwordEncoder;
        this.authenticationManager = authenticationManager;
        this.jwtService = jwtService;
    }

    public AuthResponse register(RegisterRequest request) {
        var email = request.email().trim().toLowerCase();
        if (userRepository.existsByEmailIgnoreCase(email)) {
            throw new BusinessException("E-mail já cadastrado");
        }
        var user = new User(request.name().trim(), email, passwordEncoder.encode(request.password()), Role.USER);
        userRepository.save(user);
        return response(user);
    }

    public AuthResponse login(LoginRequest request) {
        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.email(), request.password()));
        var user = userRepository.findByEmailIgnoreCase(request.email())
                .orElseThrow(() -> new BusinessException("Credenciais inválidas"));
        return response(user);
    }

    private AuthResponse response(User user) {
        return new AuthResponse(jwtService.generateToken(user), user.getId(), user.getName(), user.getEmail(), user.getRole());
    }
}

File: Francisco/francisco_prova/pw-gerenciamento-financeiro-dev/pw-gerenciamento-financeiro-dev/Front-end/src/services/authService.js
import api from '../config/axiosConfig';

class AuthService {
  async login(data) {
    const response = await api.post('/api/auth/login', data);
    return response;
  }

  async requestPasswordReset(email) {
    const response = await api.post('/api/auth/password-reset/request', { email });
    return response;
  }

  async confirmPasswordReset(token, newPassword) {
    const response = await api.post('/api/auth/password-reset/confirm', { token, newPassword });
    return response;
  }

  async register(data) {
    const response = await api.post('/api/users', data);
    return response;
  }
}

export default new AuthService();

